src/veritysetup/veritysetup.c

```diff
@@ -16,9 +16,7 @@
 #include "terminal-util.h"
 
 static uint32_t arg_activate_flags = CRYPT_ACTIVATE_READONLY;
-static char *arg_root_hash_signature = NULL;
-
-STATIC_DESTRUCTOR_REGISTER(arg_root_hash_signature, freep);
+static const char *arg_root_hash_signature = NULL;
```

Comment:
this is not OK. you must copy the string if it comes from the option string, since that one is freed when parse_options() is left.

Reply:
Is it? I think it's just called with argv[6], i.e. a static string.

Reply:
look again, i.e. the invocation of the function with "val" is with an allocated string

Reply:
→ #22942